src/adapters/tcp.rs

```diff
@@ -256,7 +264,42 @@ impl Local for LocalResource {
             TransportListen::Tcp(config) => config,
             _ => panic!("Internal error: Got wrong config"),
         };
-        let listener = TcpListener::bind(addr)?;
+
+        let socket = Socket::new(
+            match addr {
+                SocketAddr::V4 { .. } => Domain::IPV4,
+                SocketAddr::V6 { .. } => Domain::IPV6,
+            },
+            Type::STREAM,
+            Some(Protocol::TCP),
+        )?;
+        socket.set_nonblocking(true)?;
+        socket.set_reuse_address(true)?;
+
+        #[cfg(unix)]
+        if let Some(bind_device) = config.bind_device {
+            let device = CString::new(bind_device)?;
+
+            #[cfg(not(target_os = "macos"))]
+            socket.bind_device(Some(device.as_bytes()))?;
+
+            #[cfg(target_os = "macos")]
+            match NonZeroU32::new(unsafe { libc::if_nametoindex(device.as_ptr()) }) {
+                Some(index) => socket.bind_device_by_index(Some(index))?,
+                None => {
+                    return Err(io::Error::new(
+                        ErrorKind::NotFound,
+                        "Bind device interface not found",
+                    ))
+                }
+            }
+        }
+
+        socket.bind(&addr.into())?;
+        socket.listen(1024)?;
```

Comment:
Where does this magic number come from? Could it be `addr.port` instead?

Reply:
That is the backlog value copied from https://docs.rs/mio/0.8.6/src/mio/net/tcp/listener.rs.html#77 (The maximum number of _unaccepted_ TCP connections.)

Reply:
Could you add a comment like this in the code for future readers? 😇